### PR TITLE
Feat/dark mode

### PR DESCRIPTION
**Pull request by 0x1337ak**
## What does this do?

apply dark mode style 

## How did you test this?

Locally Manually Android IOS

### CODE REVIEW

**Comment by yjose** [comment]
@0x10Ayoub i did some refactoring to make the process more easy to use. 
One thing you need to complete here is to add theme logic for navigator and not for every screen, check [docs here ](https://reactnavigation.org/docs/themes/)

also look like you forgot to update the login screen as well as bottom sheet title

**Review by yjose — CHANGES_REQUESTED** [review]
looks good so far, check my comment

**Comment by yjose** [comment]
Looks good with the new colors, but the onboarding screen still isn't working correctly with the dark theme @0x10Ayoub

**Review by yjose — CHANGES_REQUESTED** [review]
Looks good with the new colors, but the onboarding screen still isn't working correctly with the dark theme @0x10Ayoub

**Comment by 0x10Ayoub** [comment]
@yjose Thanks I totally forgot it.

File: src/navigation/hooks.tsx
Comment (yjose) [inline src/navigation/hooks.tsx:1]: I prefer using use-theme-config.tsx as file name

File: src/navigation/navigation-container.tsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
+import type { Theme } from '@react-navigation/native';
 import { NavigationContainer as RNNavigationContainer } from '@react-navigation/native';
 import * as React from 'react';
 import { SafeAreaProvider } from 'react-native-safe-area-context';
 
 export const NavigationContainer = ({
   children,
+  theme,
```
Comment (yjose) [inline src/navigation/navigation-container.tsx:8]: not sure why you need to pass theme from the parent, you can use hook instead

File: src/screens/onboarding/cover.tsx
Comment (yjose) [inline src/screens/onboarding/cover.tsx:1]: Why svg instead of a simple image ?

**Review by yjose — CHANGES_REQUESTED** [review]

File: src/navigation/hooks.tsx
Comment (0x10Ayoub) [reply src/navigation/hooks.tsx:1]: yes i did refactor it

File: src/screens/onboarding/cover.tsx
Comment (0x10Ayoub) [reply src/screens/onboarding/cover.tsx:1]: @0x10Ayoub  to make it compatible with  the  dark mode

File: src/navigation/navigation-container.tsx (hunk 1 above)
Comment (0x10Ayoub) [reply src/navigation/navigation-container.tsx:8]: i want to make it  clear where we pass theme props

File: src/core/hooks/index.tsx [hunk 2]
```diff
@@ -1,2 +1,3 @@
 export * from './use-is-first-time';
 export * from './use-selected-theme';
+export * from './use-theme-config';
```
Comment (yjose) [inline src/core/hooks/index.tsx:3]: this file is related to navigation, please put  it to navigation folder

**Review by yjose — CHANGES_REQUESTED** [review]
here is some comments, also  you forgot to mention how to switching from dark, light, system theme is done, you need to add it

File: src/navigation/navigation-container.tsx (hunk 1 above)
Comment (0x10Ayoub) [reply src/navigation/navigation-container.tsx:8]: @yjose what do you think about this 👆

**Review by yjose — CHANGES_REQUESTED** [review]
One last thing, please make sure to fix capitalization while starting a new sentence and add the following video at the beginning of the Dark Mode section to showcase the feature.

https://github.com/obytes/react-native-template-obytes/assets/11137944/37620367-b639-417e-a802-c48a3d7566ba

**Review by yjose — APPROVED** [review]

**Pull request merged**